### ISSUE TITLE
CSS properties are being applied twice for text node inside Flexbox

### ISSUE THREAD

**Issue by ocerman** [servo/servo#19793]
![servo](https://user-images.githubusercontent.com/10187350/35052603-4b22cfd0-fba8-11e7-91ef-fc6ab15bc5e0.png)
```
<!DOCTYPE html>
<html>
	<head>
		<style>
			html{
				background: #BADA55;
			}
			.item{
				margin: 10px;
				height: 40px;
				color: white;
				display: inline-flex;
				border: 2px solid cyan;
				
				background: rgba(0, 0, 0, 0.7);
				transform: rotateZ(-40deg) translate(0px, 80px) scale(1.5);
				filter: invert(100%);
				font-size: 2em;
			}
		</style>
	</head>
	<body>
		<div class="item">
			Servo
		</div>
	</body>
</html>
```

**Comment by jdm** [comment]
This sounds like https://github.com/servo/servo/issues/11300, which was supposed to be fixed by https://github.com/servo/servo/pull/14078 but which was never merged.

**Comment by blacksvk** [comment]
@jdm yes, it sounds similiar, however, I tried all test cases in #[11300](https://github.com/servo/servo/issues/11300) and they were rendered correctly. This issue seems to be connected to flexbox, because it is happening when using `display: flex` or `display:inline-flex`.

**Comment by lumiscosity** [comment]
Appears to have been fixed. Tested on Arch Linux x86_64, servo built against commit id 9d29017c0dc6eec4ca7cd868cc13056e0bc0c64a, compared against Firefox 141.0, Epiphany 48.5 and Chromium 138.0.7204.168 (ordered left to right).

<img width="1736" height="394" alt="Image" src="https://github.com/user-attachments/assets/18742df5-1d48-4a1f-a89f-83826704c001" />

**Comment by jdm** [comment]
Yay!
